File: userservice/src/main/java/com/akulinski/userr8meservice/core/domain/messages/UserRateChangedMessage.java
package com.akulinski.userr8meservice.core.domain.messages;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.Set;

@Data
@AllArgsConstructor
public class UserRateChangedMessage {
  private static final String TOPIC = "STATUS_CHANGE";

  private final String  username;
  private final Set<String> followers;
  private final Double oldValue;
  private final Double newValue;
}

File: pushservice/src/main/kotlin/com/akulinski/pushservice/core/domain/messages/UserRateChangedMessage.kt
package com.akulinski.pushservice.core.domain.messages

import java.io.Serializable

class UserRateChangedMessage(
   val username: String,
   val followers: Set<String>,
   val oldValue: Double,
   val newValue: Double
) : Serializable {
  val topic = "STATUS_CHANGE"
}
